samples/dnn/text_detection.cpp

```diff
@@ -72,52 +81,82 @@ int main(int argc, char** argv)
         }
 
         blobFromImage(frame, blob, 1.0, Size(inpWidth, inpHeight), Scalar(123.68, 116.78, 103.94), true, false);
-        net.setInput(blob);
-        net.forward(outs, outNames);
+        detector.setInput(blob);
+        tick_meter.start();
+        detector.forward(outs, outNames);
+        tick_meter.stop();
 
         Mat scores = outs[0];
         Mat geometry = outs[1];
 
         // Decode predicted bounding boxes.
         std::vector<RotatedRect> boxes;
         std::vector<float> confidences;
-        decode(scores, geometry, confThreshold, boxes, confidences);
+        decode_bounding_boxes(scores, geometry, confThreshold, boxes, confidences);
 
         // Apply non-maximum suppression procedure.
         std::vector<int> indices;
         NMSBoxes(boxes, confidences, confThreshold, nmsThreshold, indices);
 
-        // Render detections.
         Point2f ratio((float)frame.cols / inpWidth, (float)frame.rows / inpHeight);
+
+        // Render text.
         for (size_t i = 0; i < indices.size(); ++i)
         {
             RotatedRect& box = boxes[indices[i]];
+            Size2f box_size = box.size;
 
             Point2f vertices[4];
             box.points(vertices);
+
             for (int j = 0; j < 4; ++j)
             {
                 vertices[j].x *= ratio.x;
                 vertices[j].y *= ratio.y;
             }
+
+            if (!model_recognition.empty())
+            {
+                Point2f box_center = box.center;
+
+                box_size.width *= ratio.x;
+                box_size.height *= ratio.y;
+
+                box_center.x *= ratio.x;
+                box_center.y *= ratio.y;
+                Mat cropped;
+                four_points_transform(frame, vertices, cropped);
+
+                cvtColor(cropped, cropped, cv::COLOR_BGR2GRAY);
+
+                Mat blob_crop = blobFromImage(cropped, 1.0/127.5, Size(), Scalar::all(127.5));
+                recognizer.setInput(blob_crop);
+
+                tick_meter.start();
+                Mat result = recognizer.forward();
+                tick_meter.stop();
+
+                std::string word_recognized = "";
+                decode_text(result, word_recognized);
+                putText(frame, word_recognized, Point2f(box_center.x - box_size.width / 2, box_center.y + box_size.height / 2),
+                    FONT_HERSHEY_SIMPLEX, 1.5, Scalar(0, 0, 255));
```

Comment:
Am I right that left top vertex is used as a start point? If so, please use `vertices` array (it contains this point).

Reply:
Fixed